src/arch/x86/tlb.cc

```diff
@@ -371,7 +377,15 @@ TLB::translate(const RequestPtr &req,
         if (m5Reg.paging) {
             DPRINTF(TLB, "Paging enabled.\n");
             // The vaddr already has the segment base applied.
-            TlbEntry *entry = lookup(vaddr);
+
+            //Appending the pcid (last 12 bits of CR3) to the
+            //page aligned vaddr
+            Process *p_temp = tc->getProcessPtr();
```

Comment:
I don't think this will work in FS mode as the emulated page tables are not used in FS mode. The only thing I need from the process is the page size to align the vaddr to the page size. I am not sure, what't the best way to do that in FS mode or maybe we can align the vaddr to a constant 4K page size?

Reply:
The answer is `X86ISA::PageShift` or `X86ISA::PageBytes` depending on if you want the shift or the size. See `src/arch/x86/traits.hh`